File: Source/Aladdin.CAPI.ANSI.Java/src/aladdin/asn1/ansi/x962/ECDomainParameters.java
package aladdin.asn1.ansi.x962;
import aladdin.asn1.*; 

////////////////////////////////////////////////////////////////////////////////
// EcpkParameters ::= CHOICE {
//      ecParameters  ECParameters,
//      namedCurve    OBJECT IDENTIFIER,
//      implicitlyCA  NULL 
// }
////////////////////////////////////////////////////////////////////////////////
public final class ECDomainParameters extends Choice
{
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(SpecifiedECDomain.class).factory(), Cast.N), 
		new ObjectInfo(new ObjectCreator(ObjectIdentifier .class).factory(), Cast.N), 
		new ObjectInfo(new ObjectCreator(Null             .class).factory(), Cast.N), 
	}; 
	// конструктор
	public ECDomainParameters() { super(info); } 
}

File: Source/Aladdin.CAPI.ANSI/ASN1/ANSI/X962/ECDomainParameters.cs
namespace Aladdin.ASN1.ANSI.X962
{
    ////////////////////////////////////////////////////////////////////////////////
    // EcpkParameters ::= CHOICE {
    //      ecParameters  ECParameters,
    //      namedCurve    OBJECT IDENTIFIER,
    //      implicitlyCA  NULL 
    // }
    ////////////////////////////////////////////////////////////////////////////////
    public class ECDomainParameters : Choice
    {
	    // информация о структуре
	    private static readonly ObjectInfo[] info = new ObjectInfo[] { 

		    new ObjectInfo(new ObjectCreator<SpecifiedECDomain>().Factory(), Cast.N), 
		    new ObjectInfo(new ObjectCreator<ObjectIdentifier >().Factory(), Cast.N), 
		    new ObjectInfo(new ObjectCreator<Null             >().Factory(), Cast.N), 
	    }; 
	    // конструктор
	    public ECDomainParameters() : base(info) {} 
    }
}
